File: mRemoteV1/Controls/ToolStripSplitButton.vb
Namespace Controls
    Public Class ToolStripSplitButton
        Inherits System.Windows.Forms.ToolStripSplitButton

        Public Overloads Property DropDown As ToolStripDropDown
            Get
                Return MyBase.DropDown
            End Get
            Set
                If MyBase.DropDown IsNot value Then
                    MyBase.DropDown = value
                    AddHandler MyBase.DropDown.Closing, AddressOf DropDown_Closing
                End If
            End Set
        End Property

        Private Sub DropDown_Closing(sender As Object, e As ToolStripDropDownClosingEventArgs)
            If Not e.CloseReason = ToolStripDropDownCloseReason.AppClicked Then Return

            Dim dropDownButtonBoundsClient As Rectangle = DropDownButtonBounds ' Relative to the ToolStripSplitButton
            dropDownButtonBoundsClient.Offset(Bounds.Location) ' Relative to the parent of the ToolStripSplitButton
            Dim dropDownButtonBoundsScreen As Rectangle =
                    GetCurrentParent().RectangleToScreen(dropDownButtonBoundsClient) ' Relative to the screen

            If dropDownButtonBoundsScreen.Contains(Control.MousePosition) Then e.Cancel = True
        End Sub

        Protected Overrides Sub OnMouseDown(e As MouseEventArgs)
            _dropDownVisibleOnMouseDown = DropDown.Visible
            MyBase.OnMouseDown(e)
        End Sub

        Protected Overrides Sub OnMouseUp(e As MouseEventArgs)
            If _dropDownVisibleOnMouseDown Then
                DropDown.Close()
            Else
                MyBase.OnMouseUp(e)
            End If
        End Sub

        Private _dropDownVisibleOnMouseDown As Boolean = False
    End Class
End Namespace

File: mRemoteV1/UI/Controls/ToolStripSplitButton.cs
using System.Drawing;
using System.Windows.Forms;

namespace mRemoteNG.UI.Controls
{
	public class ToolStripSplitButton : System.Windows.Forms.ToolStripSplitButton
	{
        public new ToolStripDropDown DropDown
		{
			get { return base.DropDown; }
			set
			{
				if (base.DropDown != value)
				{
					base.DropDown = value;
					base.DropDown.Closing += DropDown_Closing;
				}
			}
		}
			
		private void DropDown_Closing(object sender, ToolStripDropDownClosingEventArgs e)
		{
			if (e.CloseReason != ToolStripDropDownCloseReason.AppClicked)
			{
				return;
			}
				
			Rectangle dropDownButtonBoundsClient = DropDownButtonBounds; // Relative to the ToolStripSplitButton
			dropDownButtonBoundsClient.Offset(Bounds.Location); // Relative to the parent of the ToolStripSplitButton
			Rectangle dropDownButtonBoundsScreen = GetCurrentParent().RectangleToScreen(dropDownButtonBoundsClient); // Relative to the screen
				
			if (dropDownButtonBoundsScreen.Contains(Control.MousePosition))
			{
				e.Cancel = true;
			}
		}
			
		protected override void OnMouseDown(MouseEventArgs e)
		{
			_dropDownVisibleOnMouseDown = DropDown.Visible;
			base.OnMouseDown(e);
		}
			
		protected override void OnMouseUp(MouseEventArgs e)
		{
			if (_dropDownVisibleOnMouseDown)
			{
				DropDown.Close();
			}
			else
			{
				base.OnMouseUp(e);
			}
		}
			
		private bool _dropDownVisibleOnMouseDown;
	}
}
